### PR TITLE
Add extra debugging and error annotation in cmr workers

### PR DESCRIPTION
**Pull request by wallyworld**
Add extra error annotations and logging in various cross model relations workers / facades.

## QA steps

deploy a cmr and look at logs.

### CODE REVIEW

File: apiserver/common/crossmodel/crossmodel.go [hunk 1]
```diff
@@ -190,11 +190,11 @@ func handleChangedUnits(change params.RemoteRelationChangeEvent, applicationTag
 func GetOfferingRelationTokens(backend Backend, tag names.RelationTag) (string, string, error) {
 	offerName, err := backend.OfferNameForRelation(tag.Id())
 	if err != nil {
-		return "", "", errors.Trace(err)
+		return "", "", errors.Annotatef(err, "error getting offer for relation %q", tag.Id())
```
Comment (ycliuhw) [inline apiserver/common/crossmodel/crossmodel.go:193]: do we need the "error "?

**Review by ycliuhw — APPROVED** [review]

**Comment by wallyworld** [comment]
$$merge$$

File: apiserver/common/crossmodel/crossmodel.go (hunk 1 above)
Comment (wallyworld) [reply apiserver/common/crossmodel/crossmodel.go:193]: can take it or leave it - there's several other places that include it; the other common one to use is "cannot ..."

Comment (wallyworld) [reply apiserver/common/crossmodel/crossmodel.go]: removed the "error"

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**